blogposts/2022/release-3.39

```diff
@@ -0,0 +1,75 @@
+    url: https://github.com/sourcegraph/sourcegraph/pull/33866
+    category: Code Insights
+  - description: "The Code Insights commit indexer can now index commits in smaller batches to ensure it succeeds for repositories with many commits. Set the number of days per batch in the site setting `insights.commit.indexer.windowDuration`. A value of 0 (default) will disable batching."
+    url: https://github.com/sourcegraph/sourcegraph/pull/33666
+    category: Code Insights
+  - description: "Code Insights series data is now sorted by semantic version and then alphabetically to make it easier to understand data in order of earliest to latest versions."
+    url: https://github.com/sourcegraph/sourcegraph/pull/33454
+    category: Code Insights
+  - description: "We've optimized search patterns containing `and` and `not` expressions. These kinds of queries are now an order of magnitude faster. Previous cases where no results were returned due to hitting the file limit should now work and return results quickly."
+    url: https://github.com/sourcegraph/sourcegraph/pull/33308
+    category: Search
+  - description: "Built-in authentication (i.e. username and password) now supports account lockout after consecutive failed sign-in attempts. New config options have been added under `\auth.lockout\` to customize the threshold for failed attempts and length of lockouts."
+    url: https://github.com/sourcegraph/sourcegraph/pull/33999
+    category: Admin
+---
+
+Sourcegraph 3.39 is now available! In this release:
+
+## Onboard developers fast with Notebooks, now Generally Available
+
+Built with onboarding in mind, Notebooks allow you to integrate search results and Markdown to make documentation that is both effortless and doesn't get stale. Notebooks are inspired by the Jupyter Notebooks project and feature four unique block types, an intelligent notepad for creating notebooks on the fly, and deep integration with Code Search so you can always find the code you need.
+
+<video class="blog-image" title="Notebooks overview" alt="An example of using a notebook." loop autoplay muted playsinline>
+  <source src="https://user-images.githubusercontent.com/1657213/163855878-4355bcb0-5407-4307-b3f2-4581e1ab2fe2.mp4">
+</video>
+
+Web-based notebooks are the easiest way to create and share notebooks with your team or company, but we also support file-based notebooks via a special `.snb.md` file type. With file-based Notebooks, you can transform your existing documentation and augment it with Sourcegraph query blocks. You can also embed notebooks on any web page, enabling easy integration with your own knowledge management systems, so you can meet your developers where they are.
+
+Notebooks are now available for all users in Sourcegraph 3.39, and you can also explore some of our [public notebooks on Sourcegraph Cloud](https://sourcegraph.com/notebooks?tab=explore) with no account required.
```

Comment:
@iskyOS @benvenker - thoughts on explicitly calling out that Notebooks are free?

Reply:
I like that idea — can't hurt and makes it clearer that you don't need to talk to your CE for it.

```suggestion
Notebooks are now available for free for all users in Sourcegraph 3.39, and you can also explore some of our [public notebooks on Sourcegraph Cloud](https://sourcegraph.com/notebooks?tab=explore) with no account required.
```